content/blog/migrating-a-cloud-application-to-kubernetes/index.md

```diff
@@ -0,0 +1,476 @@
+---
+title: "Migrating a cloud application to Kubernetes"
+date: 2020-09-11
+meta_desc: Using Pulumi to integrate applications with Kubernetes and allow for on-demand scalability of cloud architecture and freedom of design.
+meta_image: meta.png
+authors: ["vova-ivanov"]
+tags: ["aws", "typescript", "docker", "kubernetes"]
+---
+
+In this blog post, we will explore the advantages of Kubernetes, and will demonstrate how to use it by converting and deploying our [PERN application]({{< relref "/blog/deploying-a-pern-stack-application-to-aws" >}}) to Amazon EKS. With the help of Pulumi, the process becomes greatly simplified, and allows us to focus more on the big picture of designing our cloud architecture.
+
+<!--more-->
+
+Kubernetes offers several distinct advantages over individually deploying Docker containers to the cloud. Firstly, is its robust system for minimizing downtime and rolling out updates. If a pod or node were to be shut down, a new one will be automatically allocated to take its place. Secondly, is the ability to scale and distribute your infrastructure. By simplifying containers into pods, Kubernetes allows you to effortlessly design the right cloud setup to fit your target. And lastly, is the interchangeability of cloud providers that comes with it. Since AWS, Azure, and GCE all support deploying Kubernetes, it means that you are not forced to stick with one provider, and have the option of easily switching if the need arises.
```

Comment:
```suggestion
Kubernetes offers several distinct advantages over individually deploying Docker containers to the cloud. Firstly, is its robust system for minimizing downtime and rolling out updates. If a pod or node were to be shut down, a new one will be automatically allocated to take its place. Secondly, is the ability to scale and distribute your infrastructure. By simplifying containers into pods, Kubernetes allows you to effortlessly design the right cloud setup to fit your target. And lastly, is the interchangeability of cloud providers that comes with it. Since AWS, Azure, and GCP all support deploying Kubernetes, it means that you are not forced to stick with one provider, and have the option of easily switching if the need arises.
```